scripts/prepare_network.py

```diff
@@ -59,18 +59,72 @@
 import logging
 import os
 import re
+from zipfile import ZipFile
 
+import country_converter as coco
 import numpy as np
 import pandas as pd
 import pypsa
-from _helpers import configure_logging
+import requests
+from _helpers import configure_logging, create_country_list
 from add_electricity import load_costs, update_transmission_costs
+from config_osm_data import continent_regions, world_iso
 
 idx = pd.IndexSlice
 
 logger = logging.getLogger(__name__)
 
 
+def download_emission_data():
+    try:
+        url = "https://jeodpp.jrc.ec.europa.eu/ftp/jrc-opendata/EDGAR/datasets/v60_GHG/CO2_excl_short-cycle_org_C/v60_GHG_CO2_excl_short-cycle_org_C_1970_2018.zip"
+        with requests.get(url) as rq:
+            with open("data/co2.zip", "wb") as file:
+                file.write(rq.content)
+        rootpath = os.getcwd()
+        file_path = os.path.join(rootpath, "data/co2.zip")
+        with ZipFile(file_path, "r") as zipObj:
+            zipObj.extract(
+                "v60_CO2_excl_short-cycle_org_C_1970_2018.xls", rootpath + "/data"
+            )
+        os.remove(file_path)
+        return "v60_CO2_excl_short-cycle_org_C_1970_2018.xls"
+    except:
+        logger.warning(f"Failed download resource from '{url}'.")
+        return False
+
+
+def emission_extractor(filename):
+    country_names = create_country_list(snakemake.config["countries"])
+    # data reading process
+    datapath = os.path.join(os.getcwd(), "data", filename)
+    emission_of_countries = []
+    df = pd.ExcelFile(datapath)
```

Comment:
I don't think you need to do this preprocessing.
below, it should work `read_excel(datapath, ...`

Reply:
I will try that and I think it will make code look more clean. Thank you :)